File: Scripts/Glasses.cs
using Godot;
using System;
using FSharpScripts;

public partial class Glasses : Node3D
{
	// Called when the node enters the scene tree for the first time.
	public override void _Ready()
	{
		GlassesFS.ready();
	}

	// Called every frame. 'delta' is the elapsed time since the previous frame.
	public override void _PhysicsProcess(double delta)
	{
		GlassesFS.physicsProcess((float) delta);
	}
}

File: FSharpScripts/Glasses.fs
namespace FSharpScripts

open Godot
open GlobalFunctions

module GlassesFS =
    let mutable self = Unchecked.defaultof<Node3D>
    
    let ready () = self <- getRoot().GetNode<Node3D>("WorldGenerator").GetNode<Node3D>("Glasses").GetNode<Node3D>("Model")
        
    let physicsProcess delta = self.RotateY(degToRad 45f * delta)